modules/nvidia-gpu-admin-dashboard-installing.adoc

```diff
@@ -0,0 +1,128 @@
+// Module included in the following assemblies:
+//
+// * monitoring/nvidia-gpu-admin-dashboard.adoc
+
+:_content-type: PROCEDURE
+[id="nvidia-gpu-admin-dashboard-installing_{context}"]
+= Installing the NVIDIA GPU administration dashboard
+
+You install the NVIDIA GPU plugin using Helm on the OpenShift Container Platform (OCP) Console to add GPU capabilities.
+
+The OpenShift Console NVIDIA GPU plugin works as a remote bundle for the OCP console. To run the OpenShift Console NVIDIA GPU plugin
+an instance of the OCP console must be running.
+
+
+.Prerequisites
+
+* Red Hat OpenShift 4.11+
+* NVIDIA GPU operator
+* Helm
+
+KQ - Can we provide a link to installing helm?
+
+.PROCEDURE
+
+Use the following procedure to deploy the OpenShift Console NVIDIA GPU plugin.
+
+. Add the Helm repository:
++
+[source,terminal]
+----
+$ helm repo add rh-ecosystem-edge https://rh-ecosystem-edge.github.io/console-plugin-nvidia-gpu
+----
++
+[source,terminal]
+----
+$ helm repo update
+----
+
+. Install the Helm chart in the default NVIDIA GPU operator namespace:
++
+[source,terminal]
+----
+$ helm install -n nvidia-gpu-operator console-plugin-nvidia-gpu rh-ecosystem-edge/console-plugin-nvidia-gpu
+----
++
+.Example output
++
+[source,terminal]
+----
+NAME: console-plugin-nvidia-gpu
+LAST DEPLOYED: Tue Aug 23 15:37:35 2022
+NAMESPACE: nvidia-gpu-operator
+STATUS: deployed
+REVISION: 1
+NOTES:
+View the Console Plugin NVIDIA GPU deployed resources by running the following command:
+
+$ kubectl -n nvidia-gpu-operator get all -l app.kubernetes.io/name=console-plugin-nvidia-gpu
+
+Enable the plugin by running the following commands:
+
+# check if a plugins field is specified
+$ oc get consoles.operator.openshift.io cluster --output=jsonpath="{.spec.plugins}"
+
+# if not, then run the following to enable the plugin
+$ kubectl patch consoles.operator.openshift.io cluster --patch '{ "spec": { "plugins": ["console-plugin-nvidia-gpu"] } }' --type=merge
+
+# if yes, then run the following to enable the plugin
+$ kubectl patch consoles.operator.openshift.io cluster --patch '[{"op": "add", "path": "/spec/plugins/-", "value": "console-plugin-nvidia-gpu" }]' --type=json
```

Comment:
This part has been updated and it should now look like this:

```
NOTES:
View the Console Plugin NVIDIA GPU deployed resources by running the following command:

$ kubectl -n {{ .Release.Namespace }} get all -l app.kubernetes.io/name=console-plugin-nvidia-gpu

Enable the plugin by running the following commands:

# check if a plugins field is specified
$ oc get consoles.operator.openshift.io cluster --output=jsonpath="{.spec.plugins}"

# if not, then run the following to enable the plugin
$ oc patch consoles.operator.openshift.io cluster --patch '{ "spec": { "plugins": ["console-plugin-nvidia-gpu"] } }' --type=merge

# if yes, then run the following to enable the plugin
$ oc patch consoles.operator.openshift.io cluster --patch '[{"op": "add", "path": "/spec/plugins/-", "value": "console-plugin-nvidia-gpu" }]' --type=json

# add the required DCGM Exporter metrics ConfigMap to the existing NVIDIA operator ClusterPolicy CR
$ oc patch clusterpolicies.nvidia.com gpu-cluster-policy --patch '{ "spec": { "dcgmExporter": { "config": { "name": "console-plugin-nvidia-gpu" } } } }' --type=merge
```

There is an additional step in the end of the `NOTES`, which enables the appropriate Prometheus metrics required for the NVIDIA GPU Administration Dashboard to be fully functional.

> The dashboard relies (mostly) on Prometheus metrics exposed by the [NVIDIA DCGM Exporter](https://github.com/NVIDIA/dcgm-exporter), but the default exposed metrics are not enough for the dashboard to render the required gauges. Thus, the DGCM exporter should be configured to expose a custom set of [metrics](https://github.com/rh-ecosystem-edge/console-plugin-nvidia-gpu/blob/main/deployment/console-plugin-nvidia-gpu/templates/configmap.yaml#L8). This is achieved by deploying a ConfigMap that includes those custom metrics and editing the NVIDIA operator ClusterPolicy CR to add that ConfigMap in the DCGM exporter configuration. The deployment of the ConfigMap is done by the new [version](https://github.com/rh-ecosystem-edge/console-plugin-nvidia-gpu/blob/gh-pages/console-plugin-nvidia-gpu-0.2.3.tgz) of the Console Plugin NVIDIA GPU Helm Chart, but the ClusterPolicy CR editing should be done by the user, as explained [here](https://github.com/rh-ecosystem-edge/console-plugin-nvidia-gpu/blob/main/deployment/console-plugin-nvidia-gpu/templates/NOTES.txt#L16).

Reply:
@mresvanis Is the above paragraph really in the NOTES section in the codeblock  or should I place it after the codeblock?

Reply:
The additional step in the `NOTES` section, to which I'm referring, is this one:

```
# add the required DCGM Exporter metrics ConfigMap to the existing NVIDIA operator ClusterPolicy CR
$ oc patch clusterpolicies.nvidia.com gpu-cluster-policy --patch '{ "spec": { "dcgmExporter": { "config": { "name": "console-plugin-nvidia-gpu" } } } }' --type=merge
```

The quoted paragraph in my previous comment is not in the `NOTES` section, it's an explanation in my own wording of why this step is needed, but I'm not sure whether or not we should add it to the docs. Whatever you think is best.

Reply:
@mresvanis Ok. I'll make these changes and repush the PR today. Please ensure I got this right. Note that Red Hat documentation does not allow links to GitHub sites, so I'm going to publish the metrics ConfigMap.